packages/shared/src/hooks/streaks/useReadingStreak.ts

```diff
@@ -10,13 +10,16 @@ interface UserReadingStreak {
   isEnabled: boolean;
 }
 
-export const useReadingStreak = (): UserReadingStreak => {
+export const useReadingStreak = (skipCache = false): UserReadingStreak => {
   const { user } = useAuthContext();
   const { shouldShowStreak } = useStreakExperiment();
   const { data: streak, isLoading } = useQuery(
     generateQueryKey(RequestKey.UserStreak, user),
     getReadingStreak,
-    { enabled: shouldShowStreak, staleTime: StaleTime.Default },
+    {
+      enabled: shouldShowStreak,
+      staleTime: skipCache ? null : StaleTime.Default,
+    },
```

Comment:
I think it will be better to move the alerts logic inside the hook. Like this if any of the hooks without `skipCache` passed executes first it will still have stale time. That way also all hooks will behave the same based on alerts.

Reply:
@capJavert I changed this to just invalidating the query when the milestone modal needs to be displayed, wanted to avoid polluting this hook with alert related business logic. WDYT?